source/common/http/conn_manager_impl.cc

```diff
@@ -827,8 +827,12 @@ ConnectionManagerImpl::ActiveStream::ActiveStream(ConnectionManagerImpl& connect
          "Either routeConfigProvider or (scopedRouteConfigProvider and scopeKeyBuilder) should be "
          "set in "
          "ConnectionManagerImpl.");
-  for (const AccessLog::InstanceSharedPtr& access_log : connection_manager_.config_->accessLogs()) {
-    filter_manager_.addAccessLogHandler(access_log);
+  if (Runtime::runtimeFeatureEnabled(
+          "envoy.reloadable_features.quic_defer_logging_to_ack_listener")) {
+    for (const AccessLog::InstanceSharedPtr& access_log :
+         connection_manager_.config_->accessLogs()) {
+      filter_manager_.addAccessLogHandler(access_log);
+    }
```

Comment:
why we do this?

Reply:
I see, for quic. The quick need access the logger handler list by the accessLogHandlers() list.

Then the correct way is to enhance the `accessLogHandlers()` to generate a full logger list when it's called and ensure the order of list is same with the order of logger evaluation in the log() method.